index.js

```diff
@@ -48,11 +50,47 @@ function spawnClangFormat(args, done, stdio) {
               "Consider installing it with your native package manager instead.\n";
     throw new Error(message);
   }
-  var clangFormatProcess = spawn(nativeBinary, args, {stdio: stdio});
-  clangFormatProcess.on('close', function(exit) {
-    if (exit) done(exit);
-  });
-  return clangFormatProcess;
+
+  // extract glob, if present
+  var filesGlob = args.filter(function(arg){return arg.indexOf('--glob=') !== -1;})
+                    .map(function(arg){return arg.replace('--glob=', '');})
+                    .shift();
+
+  if (filesGlob) {
+    // remove glob from arg list
+    args = args.filter(function(arg){return arg.indexOf('--glob=') === -1;});
+
+    return glob(filesGlob, function(er, files) {
+      // launch a new process for each file, up to 200 at a time
+      async.parallelLimit(
+        files.map(function(file) {
+          return function(callback) {
+            var clangFormatProcess = spawn(nativeBinary,
+                                          args.concat(file),
+                                          {stdio: stdio});
+            clangFormatProcess.on('close', function(exit) {
+              if (exit !== 0) callback(exit);
+              else callback(null, exit);
+            });
+          };
+        }),
+        200,
```

Comment:
hmm, do we have an idea how much memory each clang-format process consumes? we should be careful to set this limit such that we don't hose your machine on a big project with 200 big files.